test/test_tune.py

```diff
@@ -110,58 +110,55 @@ def autotune(cls, context: AbstractContext, param: str) -> Mapping[str, Any]:
         ret3 = self.autotune(context, 'xyz')
-        assert ret3 == tuning2
-
-    @mock.patch('katsdpsigproc.tune._close_db')
-    @mock.patch('katsdpsigproc.tune._open_db')
-    @mock.patch('katsdpsigproc.tune.KATSDPSIGPROC_TUNE_MATCH', 'nearest')
-    def test_nearest(self, open_db_mock: mock.Mock, close_db_mock: mock.Mock,
-                     conn: sqlite3.Connection) -> None:
-        # First populate the database with one config
-        open_db_mock.return_value = conn
-        tuning = {'a': 1, 'b': 2}
-        context1 = mock.NonCallableMock()
-        context1.device.driver_version = 'mock version'
-        context1.device.platform_name = 'mock platform'
-        context1.device.name = 'mock device'
-        self.autotune_mock.return_value = tuning
-        ret = self.autotune(context1, 'xyz')
-        assert ret == tuning
-
-        # Test incremental fallback on [version | platform | device]
-        context1.device.driver_version = 'another mock version'
-        ret2 = self.autotune(context1, 'xyz')
-        context1.device.platform_name = 'another mock platform'
-        ret3 = self.autotune(context1, 'xyz')
-        context1.device.name = 'another mock device'
-        ret4 = self.autotune(context1, 'xyz')
-        assert ret == ret2 == ret3 == ret4
-
-        # Now re-run with a different config, and make sure it uses the
-        # cached value.
-        context2 = mock.NonCallableMock()
-        context2.device.driver_version = 'mock version 2'
-        context2.device.platform_name = 'mock platform 2'
-        context2.device.name = 'mock device 2'
-        self.autotune_mock.side_effect = RuntimeError
-        ret = self.autotune(context2, 'xyz')
-        assert ret == tuning
+        assert ret3 == another_tuning
+
+        # Now test incremental fallback on device  [ driver version | platform | name ] when
+        # KATSDPSIGPROC_TUNE_MATCH is set to 'nearest'
+        with mock.patch('katsdpsigproc.tune.KATSDPSIGPROC_TUNE_MATCH', 'nearest'):
+            # Re-run with different configurations, and make sure it always uses cached values.
+            self.autotune_mock.side_effect = RuntimeError
+
+            context.device.driver_version = 'abc'
+            context.device.platform_name = 'mock platform'
+            context.device.name = 'mock device'
+            ret4 = self.autotune(context, 'xyz')
+            assert ret4 == tuning
+
+            context.device.driver_version = 'abc'
+            context.device.platform_name = 'abc'
+            context.device.name = 'another device'
+            ret5 = self.autotune(context, 'xyz')
+            assert ret5 == another_tuning
+
+            # If there is no match in the database, any tuning will suffice
+            context.device.driver_version = 'abc'
+            context.device.platform_name = 'abc'
+            context.device.name = 'abc'
+            ret6 = self.autotune(context, 'xyz')
+            assert ret6 == tuning or another_tuning
```

Comment:
I don't think this means what you think it means - it cannot ever fail because `another_tuning` is truthy. I assume you intended something like
```suggestion
            assert ret6 in {tuning, another_tuning}
```

or
```suggestion
            assert ret6 == tuning or ret6 == another_tuning
```

Reply:
Thanks for catching this.